KIC 3.0: Rewrite annotation (#6422)
Kong/docs.konghq.com: app/_src/kubernetes-ingress-controller/guides/requests/rewrite-annotation.md

purpose: |
  What is the rewrite annotation and how does it work?
---

{{ site.kic_product_name }} provides the `konghq.com/rewrite` annotation to customize the request path before it is sent to the upstream service.

The annotation can be used on `Ingress` and `HTTPRoute` resources, and configures a [request-transformer](/hub/kong-inc/request-transformer/) plugin within Kong when added to a route.

> The following definition creates a route that matches the path `/users/(\w+)` and rewrites it to `/requests/users_svc/$1` before sending the request upstream.

Reviewer: ```suggestion
This definition creates a route that matches the path `/users/(\w+)` and rewrites it to `/requests/users_svc/$1` before sending the request upstream.
```

{% include /md/kic/http-test-routing-resource.md kong_version=page.kong_version path='/users/(\w+)' name='user' service='users' port='80' skip_host=true route_type='RegularExpression' no_results=true annotation_rewrite="/requests/users_svc/$1" %}

> This is equivalent to the following plugin configuration:

Reviewer: ```suggestion
Alternatively, you can define this in a plugin configuration.
```
